Add information about using GQIDMS outside GQI extension lifecycle (#6322)
SkylineCommunications/dataminer-docs: develop/devguide/GQI_Extensions/GQI_Extensions_Best_Practices.md

- [Do not use Skyline.DataMiner.Automation](#do-not-use-skylinedataminerautomation)
- [Only use 64-bit assembly references](#only-use-64-bit-assembly-references)
- [Make your ad hoc data source scalable](#make-your-ad-hoc-data-source-scalable)

> - [Do not use GQIDMS outside the request lifecycle](#do-not-use-gqidms-outside-the-request-lifecycle)

Reviewer: ```suggestion
- [Do not use GQIDMS outside the extension lifecycle](#do-not-use-gqidms-outside-the-request-lifecycle)
```

> [!TIP]
> For an example of how to build a highly scalable data source capable of serving many concurrent users, see [Scaling an ad hoc data source](xref:Scaling_Ad_hoc_Data_Source).

> ## Do not use GQIDMS outside the request lifecycle

Reviewer: ```suggestion
## Do not use GQIDMS outside the extension lifecycle
```

> `GQIDMS` should only be used during the lifecycle of a single GQI request. After [OnDestroy](xref:GQI_IGQIOnDestroy) is invoked, the `GQIDMS` instance for that request is cleaned up. Any in-progress or new requests sent through that instance will be cancelled.

Reviewer: ```suggestion
`GQIDMS` can only be used during the lifecycle of the extension instance. After the [OnDestroy](xref:GQI_IGQIOnDestroy) life cycle method is invoked, the `GQIDMS` object associated with the extension instance is cleaned up. Any in-progress or new requests made through that `GQIDMS` object will be cancelled.
```
